test/dataloader2/test_dataloader2.py

```diff
@@ -592,6 +603,37 @@ def test_dispatching_worker_determinism(self, ctx):
         dl.seed(321)
         self.assertNotEqual(res, list(dl) + list(dl))
 
+    @mp_ctx_parametrize
+    def test_non_replicable_datapipe(self, ctx) -> None:
+        r"""
+        For the pipeline with non-replicable DataPipe, make sure
```

Comment:
No, it can't. Like I said, `ShardingRRDispatchDP` labels the prior graph non-shardable and other non-shardable DataPipe labels itself and subsequent graph non-shardable.